libvast/src/experimental_table_slice_builder.cpp

```diff
@@ -746,8 +809,10 @@ std::shared_ptr<arrow::DataType> make_experimental_type(const type& t) {
   return caf::visit(f, t);
 }
 
+namespace {
+
 // NOLINTNEXTLINE(misc-no-recursion)
-type make_vast_type(const arrow::DataType& arrow_type) {
+type make_vast_type_int(const arrow::DataType& arrow_type) {
```

Comment:
I was initially facing compiler problems with just `make_vast_type`. Now I renamed it back and everything works, so I'm no longer able to justify the renaming. I'll revert the name change and add some docs as suggested.

Reply:
I assume this was short for "internal" then? We usually use `_impl` suffix or a `do_` prefix for those. There's also a neat trick with recursive lambdas that you can use to define the recursive helper function inline, which would probably make this even easier to read.

Reply:
I've seen compiler errors (`clangd` complains about "no matching function"...). Renaming back (removing `_int` suffix) works for gcc, but I assume that if `clangd` doesn't accept it, it will probably fail in our CI as well.

I can see more wordy function names like `make_vast_type_from_arrow_datatype` but since this is just a local function, I'm not actually sure that's better.

I'll add comments as suggested.